reducers/studyspacesReducer.js

```diff
@@ -32,7 +32,17 @@ export const initialState = {
 const updateStudyspaces = (studyspaces, id, newSpace) => {
   const idx = studyspaces.findIndex(s => s.id === id);
   const newStudyspaces = studyspaces.filter(s => s.id !== id);
-  newStudyspaces.splice(idx, 0, newSpace);
+  const oldStudySpace = studyspaces[idx];
+  newStudyspaces.splice(idx, 0, {
+    ...newSpace,
+    maps: oldStudySpace.maps.map(oldMap => {
+      const newMap = newSpace.maps.filter(m => m.id === oldMap.id)[0];
```

Comment:
Is there ever a situation where this array might have zero items in and cause an index error with the array?

Reply:
I suppose it is, if the API returns a survey with no maps or differing maps for a given survey from different endpoints. But this being JS, there are no index errors: `[][0]` is `undefined`.

And spreading `undefined` returns `{}`, which is fine.

Reply:
I am glad that this won't cause an issue, but wtf JavaScript that literally makes no sense 😂 it's convenient here though